### PR TITLE
Add Swift extractor to Code Scanning

### PR DESCRIPTION
**Pull request by calumgrant**

### CODE REVIEW

**Comment by criemen** [comment]
Search for `bazel` in https://docs.github.com/en/enterprise-cloud@latest/code-security/codeql-cli/getting-started-with-the-codeql-cli/preparing-your-code-for-codeql-analysis for the options you'll want to use

**Comment by redsun82** [comment]
hmm [the error is weird](https://github.com/github/codeql/actions/runs/8435368260/job/23100607898#step:7:115) (and as you can see it's not really related to the options you removed). Might be a bug with abseil's own bazel configuration. Let me look.

**Comment by redsun82** [comment]
maybe try to update abseil to the latest version with
```
bazel_dep(name = "abseil-cpp", version = "20240116.1")
```
in `MODULE.bazel`. You can probably revert the last commit.

**Comment by calumgrant** [comment]
The build command worked once so I presumed that whatever I changed was related to those changes!

**Comment by redsun82** [comment]
I could reproduce this locally with `--spawn_strategy=local`. It's pretty weird that a more lax sandboxing leads to that error, but it's probably a bug with `abseil`. I'll report it. In the meantime I'd suggest we go without the suggested flags.

**Comment by criemen** [comment]
Yeah, if this causes problems, you can drop everything besides `--spawn_strategy=local`, as long as we execute this on the standard ephemeral action runners.

**Comment by calumgrant** [comment]
I get the same problems with `--spawn_strategy=local` on a Linux machine. But we DO need this as we only seeing 16 compilation units in telemetry, which isn't right.

**Comment by redsun82** [comment]
hm, ok. I couldn't reproduce on a standalone `abseil` checkout, so something's off with the module import, I have an idea for a workaround, let me check

**Comment by criemen** [comment]
There's a bazel bug/design shortcoming where one header shadows another when the sandbox is disabled, thus leading to compile errors (this requires two headers of the same name). We ran into this at a customers, too. Not sure if this is what's happening with abseil, though.

**Comment by redsun82** [comment]
as a workaround, you can build `@absl//absl/strings` with the normal strategy, and then the extractor with the local one. Analysis will miss the absl strings library, but that could be good enough?

```bash
bazel build @absl//absl/strings
bazel run //swift:create-extractor-pack --spawn_strategy=local
```

**Comment by criemen** [comment]
Great find! In that case, I'd like to bring back all the options, and document why we disable the layering check here.

**Comment by redsun82** [comment]
aaand there's a bug already open: https://github.com/bazelbuild/bazel/issues/21592

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
       uses: github/codeql-action/init@main
       # Override language selection by uncommenting this and choosing your languages
       with:
-        languages: csharp
+        languages: csharp, cpp
```
Comment (henrymercer) [inline .github/workflows/codeql-analysis.yml:43]: Can't comment on the Bazel changes, but consider matrixing the analysis over languages (e.g. see the template here https://github.com/actions/starter-workflows/blob/main/code-scanning/codeql.yml).  This is what we recommend to customers as it means the analyses can run in parallel, and if one fails it doesn't disrupt the other.

Comment (henrymercer) [reply .github/workflows/codeql-analysis.yml:43]: I think when no category is specified we fall back to the workflow name, job name, and matrix parameters e.g. here `.github/workflows/codeql-analysis.yml:CodeQL-Build`.  So actually matrixing the build would create new histories when it comes to alert matching.  I think that's probably a price we're willing to pay, but it's less of a clear cut win.

Comment (adityasharad) [reply .github/workflows/codeql-analysis.yml:43]: Ah my mistake, wrong way round. Specifying the category when we do that is probably worthwhile then.

Comment (calumgrant) [reply .github/workflows/codeql-analysis.yml:43]: How do you feel about creating a second workflow file for Swift analysis? The build commands are completely different and independent.

**Comment by calumgrant** [comment]
I created a second PR https://github.com/github/codeql/pull/16071 where the C++ analysis is performed in a separate workflow file. If this approach is better, we can close this one.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (redsun82) [reply .github/workflows/codeql-analysis.yml:43]: hmm, seems like a good idea to me! Then it shouldn't disrupt the current analysis state in any way

**Comment by calumgrant** [comment]
Closed in favour of https://github.com/github/codeql/pull/16071

**Pull request closed without merging**